apps/ledger-live-mobile/src/screens/WalletCentricSections/Allocations.tsx

```diff
@@ -51,13 +55,25 @@ const Allocations = () => {
   }, [navigation]);
 
   const distributionListFormatted = useMemo(() => {
-    if (distribution.list.length <= NUMBER_MAX_ALLOCATION_ASSETS_TO_DISPLAY) {
+    const displayedCurrencies = distribution.list.filter(asset => {
+      return asset.currency.type === "TokenCurrency"
+        ? !blacklistedTokenIds.includes(asset.currency.id)
+        : true;
+    });
+
+    // if there is less than NUMBER_MAX_ALLOCATION_ASSETS_TO_DISPLAY and that there are no blacklisted tokens
+    if (
+      distribution.list.length <= NUMBER_MAX_ALLOCATION_ASSETS_TO_DISPLAY &&
+      displayedCurrencies.length === distribution.list.length
```

Comment:
Can be simplified with `if (displayedCurrencies.length <= NUMBER_MAX_ALLOCATION_ASSETS_TO_DISPLAY)`

Reply:
`displayedCurrencies.length` can be less than `NUMBER_MAX_ALLOCATION_ASSETS_TO_DISPLAY ` and less than `distribution.list.length` as well